cache.go

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright 2022 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2021 cyhone
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+* This file may have been modified by CloudWeGo authors. All CloudWeGo
+* Modifications are Copyright 2022 CloudWeGo Authors.
+*/
+
+package cache
+
+import (
+	"cache/persist"
+	"context"
+	"encoding/gob"
+	"net/http"
+	"net/url"
+	"sort"
+	"strings"
+	"time"
+	"unsafe"
+
+	"github.com/cloudwego/hertz/pkg/app"
+	"github.com/cloudwego/hertz/pkg/protocol"
+	"golang.org/x/sync/singleflight"
+)
+
+// Strategy the cache strategy
+type Strategy struct {
+	CacheKey string
+
+	// CacheStore if nil, use default cache store instead
+	CacheStore persist.CacheStore
+
+	// CacheDuration
+	CacheDuration time.Duration
+}
+
+// GetCacheStrategyByRequest User can use this function to design custom cache strategy by request.
+// The first return value bool means whether this request should be cached.
+// The second return value Strategy determine the special strategy by this request.
+type GetCacheStrategyByRequest func(c *app.RequestContext) (bool, Strategy)
```

Comment:
use func(ctx context.Context , c *app.RequestContext)

Reply:
i got it, thx